File: lib/models/feishu_public_config.dart
class FeishuPublicConfig {
  final bool enabled;
  final String enterpriseInviteUrl;
  final String notice;

  const FeishuPublicConfig({
    this.enabled = false,
    this.enterpriseInviteUrl = '',
    this.notice = '',
  });

  bool get hasInviteLink => enterpriseInviteUrl.trim().isNotEmpty;

  factory FeishuPublicConfig.fromJson(Map<String, dynamic>? json) {
    if (json == null) return const FeishuPublicConfig();
    return FeishuPublicConfig(
      enabled: json['enabled'] as bool? ?? false,
      enterpriseInviteUrl:
          (json['enterprise_invite_url'] as String?)?.trim() ?? '',
      notice: (json['notice'] as String?)?.trim() ?? '',
    );
  }
}

File: backend/utils/feishu_public_config.go
package utils

import (
	"strings"

	"github.com/spf13/viper"
)

// FeishuPublicConfig 给前端的飞书说明（无需登录）。
type FeishuPublicConfig struct {
	Enabled             bool   `json:"enabled"`
	EnterpriseInviteURL string `json:"enterprise_invite_url,omitempty"`
	Notice              string `json:"notice,omitempty"`
}

// GetFeishuPublicConfig 飞书能力为可选；机器人仅企业内可用。
func GetFeishuPublicConfig() FeishuPublicConfig {
	notice := strings.TrimSpace(viper.GetString("feishu.enterprise_notice"))
	if notice == "" {
		notice = "自建应用机器人仅可向企业内成员发送 IM 通知。如需接收角色卡等推送，可申请加入企业（非强制）。"
	}
	return FeishuPublicConfig{
		Enabled:             viper.GetBool("feishu.enabled"),
		EnterpriseInviteURL: strings.TrimSpace(viper.GetString("feishu.enterprise_invite_url")),
		Notice:              notice,
	}
}
